packages/angular_devkit/build_angular/src/protractor/index2.ts

```diff
@@ -0,0 +1,148 @@
+import { Schema as ProtractorBuilderOptions } from './schema';
+
+function runProtractor(root: string, options: ProtractorBuilderOptions): Promise<BuilderOutput> {
+  const additionalProtractorConfig: Partial<ProtractorBuilderOptions> = {
+    elementExplorer: options.elementExplorer,
+    baseUrl: options.baseUrl,
+    specs: options.specs && options.specs.length ? options.specs : undefined,
+    suite: options.suite,
+  };
+
+  // TODO: Protractor manages process.exit itself, so this target will allways quit the
+  // process. To work around this we run it in a subprocess.
+  // https://github.com/angular/protractor/issues/4160
+  return runModuleAsObservableFork(
+    root,
+    'protractor/built/launcher',
+    'init',
+    [resolve(root, options.protractorConfig), additionalProtractorConfig],
+  ).toPromise() as Promise<BuilderOutput>;
+}
+
+async function updateWebdriver() {
+  // The webdriver-manager update command can only be accessed via a deep import.
+  const webdriverDeepImport = 'webdriver-manager/built/lib/cmds/update';
+  const importOptions = [
+    // When using npm, webdriver is within protractor/node_modules.
+    `protractor/node_modules/${webdriverDeepImport}`,
+    // When using yarn, webdriver is found as a root module.
+    webdriverDeepImport,
+  ];
+
+  let path;
+  for (const importOption of importOptions) {
+    try {
+      path = require.resolve(importOption);
+    } catch (error) {
+      if (error.code !== 'MODULE_NOT_FOUND') {
+        throw error;
+      }
+    }
+  }
+
+  if (!path) {
+    throw new Error(tags.stripIndents`
+      Cannot automatically find webdriver-manager to update.
+      Update webdriver-manager manually and run 'ng e2e --no-webdriver-update' instead.
+    `);
+  }
+
+  // tslint:disable-next-line:max-line-length no-implicit-dependencies
+  const webdriverUpdate = await import(path) as typeof import ('webdriver-manager/built/lib/cmds/update');
+
+  // run `webdriver-manager update --standalone false --gecko false --quiet`
+  // if you change this, update the command comment in prev line
+  return webdriverUpdate.program.run({
+    standalone: false,
+    gecko: false,
+    quiet: true,
+  } as unknown as JSON);
+}
+
+export async function execute(
+  options: ProtractorBuilderOptions,
+  context: BuilderContext,
+): Promise<BuilderOutput> {
+  // ensure that only one of these options is used
+  if (options.devServerTarget && options.baseUrl) {
+    throw new Error(tags.stripIndents`
+    The 'baseUrl' option cannot be used with 'devServerTarget'.
+    When present, 'devServerTarget' will be used to automatically setup 'baseUrl' for Protractor.
+    `);
+  }
+
+  if (options.webdriverUpdate) {
+    try {
+      await updateWebdriver();
+    } catch (error) {
+      context.reportStatus('Error: ' + error);
+
+      return { success: false };
+    }
+  }
+
+  let baseUrl;
+  let server;
+  if (options.devServerTarget) {
+    const parts = (options.devServerTarget as string).split(':');
```

Comment:
Please use the new `targetFromTargetString()` function. It will throw if it's not in the "proper format".

Also, is `as string` necessary here? If TypeScript is confused by the type (it happened to me) you might be able to fix it with `options.devServerTarget !== undefined` in the if above).

Reply:
Done.

The cast was left over from a copy and paste.  It's not needed.